From: "Barak A. Pearlmutter" <barak+git@cs.nuim.ie>
Date: Mon, 22 Jul 2013 19:58:13 +0200
Subject: allow autoreconf force

Forwarded: no

Recombobulate slightly to allow full regeneration via the command
$ autoreconf --force

- move src/microcode/aclocal.m4 to src/microcode/mit_scheme_native_code.m4

- use single shared directory for locally-defined macros (src/microcode/ for now)

The necessary local m4 macro file is explicitly included.
This is necessary because autoconf is not using automake or libtools,
so  AC_CONFIG_MACRO_DIR([.../microcode]) won't work.  Gulp.
---
 src/configure.ac                                        | 4 ++++
 src/microcode/configure.ac                              | 4 ++++
 src/microcode/{aclocal.m4 => mit_scheme_native_code.m4} | 0
 3 files changed, 8 insertions(+)
 rename src/microcode/{aclocal.m4 => mit_scheme_native_code.m4} (100%)

diff --git a/src/configure.ac b/src/configure.ac
index 31dbf5d..81c8254 100644
--- a/src/configure.ac
+++ b/src/configure.ac
@@ -4,6 +4,10 @@ AC_INIT([MIT/GNU Scheme], [12.1], [bug-mit-scheme@gnu.org], [mit-scheme])
 AC_CONFIG_SRCDIR([microcode/boot.c])
 AC_CONFIG_AUX_DIR([microcode])
 AC_CONFIG_HEADERS([dummy-config.h])
+# Would set m4 macro dir, but only works if automake or libtools in use.
+# Instead we include directly.
+# AC_CONFIG_MACRO_DIR([microcode])
+m4_include([microcode/mit_scheme_native_code.m4])
 AC_PROG_MAKE_SET
 
 AC_COPYRIGHT(
diff --git a/src/microcode/configure.ac b/src/microcode/configure.ac
index 4fde1a4..0d98037 100644
--- a/src/microcode/configure.ac
+++ b/src/microcode/configure.ac
@@ -2,6 +2,10 @@ dnl Process this file with autoconf to produce a configure script.
 
 AC_INIT([MIT/GNU Scheme], [12.1], [bug-mit-scheme@gnu.org], [mit-scheme])
 AC_CONFIG_SRCDIR([boot.c])
+# Would set m4 macro dir, but only works if automake or libtools in use.
+# Instead we include directly.
+# AC_CONFIG_MACRO_DIR([../microcode])
+m4_include([../microcode/mit_scheme_native_code.m4])
 AC_CONFIG_HEADERS([config.h])
 AC_PROG_MAKE_SET
 
diff --git a/src/microcode/aclocal.m4 b/src/microcode/mit_scheme_native_code.m4
similarity index 100%
rename from src/microcode/aclocal.m4
rename to src/microcode/mit_scheme_native_code.m4
